src/test/java/com/es/phoneshop/service/impl/DefaultCartServiceTest.java

```diff
@@ -0,0 +1,89 @@
+package com.es.phoneshop.service.impl;
+
+import com.es.phoneshop.dao.impl.ArrayListProductDao;
+import com.es.phoneshop.exception.OutOfStockException;
+import com.es.phoneshop.model.cart.Cart;
+import com.es.phoneshop.model.cart.CartItem;
+import com.es.phoneshop.model.product.Product;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Spy;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpSession;
+import java.util.ArrayList;
+
+import static junit.framework.TestCase.assertEquals;
+import static org.mockito.ArgumentMatchers.*;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+public class DefaultCartServiceTest {
+    private static final String CART_ATTRIBUTE = "cart_" + DefaultCartService.class;
+    @Mock
+    private HttpServletRequest request;
+    @Mock
+    private HttpSession session;
+    @Spy
+    private Cart cart;
+    @Mock
+    private Product product;
+    @InjectMocks
+    private ArrayListProductDao productDao;
+    @Spy
+    private ArrayList<CartItem> cartItems;
+    @InjectMocks
+    private DefaultCartService cartService;
+
+    @Before
+    public void setup() {
+        cartService = DefaultCartService.getInstance();
+        productDao = ArrayListProductDao.getInstance();
+        productDao.getProductList().add(product);
```

Comment:
Compilation error, seriously? Also, it's the test of CartServiceTest. ProductDao must be a mock here, getProductList method stubbing is not required here